test/parallel/xyk-pallet.API.pool.deactivatedPool.test.ts

```diff
@@ -0,0 +1,316 @@
+        testUser1,
+        Xyk.createPool(
+          MGA_ASSET_ID,
+          defaultCurrencyValue,
+          token1,
+          defaultCurrencyValue
+        )
+      ),
+      Sudo.sudoAs(
+        testUser1,
+        Xyk.burnLiquidity(MGA_ASSET_ID, token1, defaultCurrencyValue)
+      )
+    );
+
+    liquidityId = await getLiquidityAssetId(MGA_ASSET_ID, token1);
+    const deactivatedPoolBalance = await getBalanceOfPool(MGA_ASSET_ID, token1);
+
+    expect(deactivatedPoolBalance[0][0]).bnEqual(BN_ZERO);
+  });
+
+  test("WHEN another user tries to create an equal pool THEN error returns", async () => {
+    await createPool(
+      testUser2.keyRingPair,
+      MGA_ASSET_ID,
+      defaultCurrencyValue,
+      token1,
+      defaultCurrencyValue
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+      expect(eventResponse.data).toEqual("PoolAlreadyExists");
+    });
+  });
+
+  test("WHEN another user tries to mint liquidity in the pool THEN user can do this", async () => {
+    await mintLiquidity(
+      testUser2.keyRingPair,
+      MGA_ASSET_ID,
+      token1,
+      defaultCurrencyValue
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+    const deactivatedPoolBalance = await getBalanceOfPool(MGA_ASSET_ID, token1);
+
+    expect(deactivatedPoolBalance[0][0]).bnGt(BN_ZERO);
+  });
+
+  test("WHEN the user mints liquidity in the pool again THEN liquidity IDs are equal", async () => {
+    await mintLiquidity(
+      testUser2.keyRingPair,
+      MGA_ASSET_ID,
+      token1,
+      defaultCurrencyValue
+    ).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+    });
+
+    const liquidityIdAfter = await getLiquidityAssetId(MGA_ASSET_ID, token1);
+
+    expect(liquidityIdAfter).bnEqual(liquidityId);
+  });
+
+  test("WHEN the user tries to swap/multiswap tokens on the deactivated pool THEN error returns", async () => {
+    let swapError: any = [];
+    try {
+      await sellAsset(
+        testUser1.keyRingPair,
+        MGA_ASSET_ID,
+        token1,
+        defaultCurrencyValue,
+        new BN(1)
+      );
+    } catch (error) {
+      swapError = error;
+    }
+    expect(swapError.data).toEqual(
+      "1010: Invalid Transaction: The swap prevalidation has failed"
+    );
+  });
+
+  test("WHEN the user tries to compound reward on a deactivated pool THEN error returns", async () => {
+    await compoundRewards(testUser1, liquidityId).then((result) => {
+      const eventResponse = getEventResultFromMangataTx(result);
+      expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+      expect(eventResponse.data).toEqual("NotAPromotedPool");
```

Comment:
plz, promote the pool to test the right test - scenario.

Reply:
Done